index.bs

```diff
@@ -4684,26 +4702,47 @@ To <dfn>parse URL pattern</dfn>, given |pattern|:
      <dd>|url|'s [=url/host=] if |has hostname| is true, or null otherwise.
      <dt>port
      <dd>
-     1. If |url|'s [=url/scheme=] [=is special=] and |url|'s
-   [=url/scheme=]'s [=default port=] is not null, and |url|'s [=url/port=] is
-   null or is equal to [=url/scheme=]'s [=default port=]:
+     1. If |has port| is false:
 
-       1. The empty string.
+       1. null.
 
-     1. Otherwise, if |url|'s [=url/port=] is not null:
+     1. Otherwise:
 
-       1. [=Serialize an integer=] with |url|'s [=url/port=].
+       1. If |url|'s [=url/scheme=] [=is special=] and |url|'s [=url/scheme=]'s
+          [=default port=] is not null, and |url|'s [=url/port=] is null or is
+          equal to [=url/scheme=]'s [=default port=]:
 
-     1. Otherwise:
+         1. The empty string.
 
-       1. null.
+       1. Otherwise, if |url|'s [=url/port=] is not null:
+
+         1. [=Serialize an integer=] with |url|'s [=url/port=].
+
+       1. Otherwise:
+
+         1. null.
 
      <dt>pathname
-     <dd>The result of running the [=URL path serializer=] with |url|, if
+     <dd>
+     1. If |has pathname| is false:
+
+       1. null.
+
+     1. Otherwise:
+
+       1. The result of running the [=URL path serializer=] with |url|, if
          |url|'s [=url/path=] is not the empty string and is not [=list/empty=],
          or null otherwise.
+
      <dt>search
-     <dd>|url|'s [=url/query=].
+     <dd>
+     1. If |has search| is false:
+
+       1. null.
+
+     1. Otherwise:
+
+       1. The empty string if |url|'s [=url/path=] is null, or |url|'s [=url/query=] otherwise.
```

Comment:
Why do we set the empty string if the path is null?

Reply:
The idea is to match the [URLPattern's spec match algorithm](https://wicg.github.io/urlpattern/#match), which defaults to an empty string if the url's search is null. 

Basically this algorithm will never differentiate between empty string search or no search, so I always turn null into empty string, both here, and when reading the components of the URL to match.

Reply:
I missed that this was `path` instead of `query` should be fixed now!